### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by SweetOps**
## What 

* Update README
* Add travis
* Utilize all attributes of `aws_efs_file_system`
* Fix `SG` for prevent recreation of EFS
* Update nested modules 

## Why

* Module should be more usable

### CODE REVIEW

**Comment by SweetOps** [comment]
screenshot 2017-11-27 12 01 35

File: README.md [hunk 1]
```diff
@@ -7,47 +7,52 @@ Terraform module to provision an AWS [`EFS`](https://aws.amazon.com/efs/) Networ
 
 Include this repository as a module in your existing terraform code:
 
-```hcl
+```terraform
```
Comment (aknysh) [inline README.md]: GitHub does not support `terraform`, it supports `hcl`

**Comment by osterman** [comment]
Let me know @SweetOps if/when ready for re-review.

**Comment by aknysh** [comment]
will close it for now since many features have been added already

**Pull request closed without merging**